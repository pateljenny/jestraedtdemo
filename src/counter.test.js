import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
// import renderer from 'react-test-renderer';

import Counter from './counter';

Enzyme.configure({ adapter: new Adapter() });

describe('Counter Component Test', () => {
    // it('matches the snapshot', () => {
    //     const tree = renderer.create(<Counter />).toJSON();
    //     expect(tree).toMatchSnapshot();
    // })
    it('starts with count = 0', () => {
        const wrapper = shallow(<Counter />)
        let output = wrapper.find('p').text();
        expect(output).toBe('Current Count Value: 0')
    })

    it('can increment count on increment button click', () => {
        const wrapper = shallow(<Counter />);
        const incrementBtn = wrapper.find('button.increment');
        incrementBtn.simulate('click');
        let output = wrapper.find('p').text();
        expect(output).toBe('Current Count Value: 1');
    })

    it('can decrement count on decrement button click', () => {
        const wrapper = shallow(<Counter />);
        const decrementBtn = wrapper.find('button.decrement');
        decrementBtn.simulate('click');
        let output = wrapper.find('p').text();
        expect(output).toBe('Current Count Value: -1');
    })
})

