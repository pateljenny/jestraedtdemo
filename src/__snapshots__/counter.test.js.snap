// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Test matches the snapshot 1`] = `
<div>
  <p>
    Current Count :
    0
  </p>
  <button
    onClick={[Function]}
  >
    Increment Count
  </button>
</div>
`;
